### PR TITLE
Code Review Groups: rename components, update styling, add containers

### PR DESCRIPTION
**Pull request by bencodeorg**
1. Renames existing `CodeReviewGroupMember` and `CodeReviewGroup` component to `Student` and `StudentGroup`, respectively. More info on naming in [this PR](https://github.com/code-dot-org/code-dot-org/pull/43175).
2. Splits code review groups manager into two components: a `UnassignedStudentsPanel` for students who have not been assigned to a code review groups, and an `AssignedStudentsPanel` that lists all existing groups.
3. Adds some styling to get closer to our desired mocks, but will leave final touches to additional UI work we have planned this sprint. Got tired of styling :). Included to-dos linked to Jira tasks in appropriate places.

image

## Testing story

Tested manually via Storybook.

### CODE REVIEW

File: apps/src/templates/codeReviewGroups/AssignedStudentsPanel.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import i18n from '@cdo/locale';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import JavalabButton from '@cdo/apps/javalab/JavalabButton';
+import StudentGroup from './StudentGroup';
+import color from '@cdo/apps/util/color';
+import {HEADER_STYLE, BUTTON_STYLE} from './UnassignedStudentsPanel';
+
+export default function AssignedStudentsPanel({groups, onCreateGroupClick}) {
```
Comment (sanchitmalhotra126) [inline apps/src/templates/codeReviewGroups/AssignedStudentsPanel.jsx:10]: Question for @madelynkasula - do we have a preference for using arrow functions for functional components? (regardless though, nice functional component!)

File: apps/src/templates/codeReviewGroups/Student.jsx [hunk 2]
```diff
@@ -41,24 +43,22 @@ export default function CodeReviewGroupMember({followerId, name, index}) {
...
-  borderRadius: '30px',
-  color: 'white',
+  color: color.dark_charcoal,
   width: 'auto',
   height: '20px',
+  border: `1px solid ${color.lighter_gray}`,
 
   // change background colour if dragging
-  background: isDragging ? 'navy' : '#0094CA',
+  background: isDragging ? 'navy' : 'white',
```
Comment (sanchitmalhotra126) [inline apps/src/templates/codeReviewGroups/Student.jsx:61]: nit: can use `color.white` here

File: apps/src/templates/codeReviewGroups/UnassignedStudentsPanel.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  height: 54,
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  padding: '0 5px',
+  border: `1px solid ${color.lighter_gray}`,
+  background: color.lightest_gray
+};
+
+export const BUTTON_STYLE = {
```
Comment (sanchitmalhotra126) [inline apps/src/templates/codeReviewGroups/UnassignedStudentsPanel.jsx:48]: Might be nice to put these in some common `styleConstants` or `codeReviewGroupsStyleConstants` file?

**Review by sanchitmalhotra126 — APPROVED** [review]
Nice, looks good! Couple of nits but nothing major

File: apps/src/templates/codeReviewGroups/AssignedStudentsPanel.jsx (hunk 1 above)
Comment (maddiedierker) [reply apps/src/templates/codeReviewGroups/AssignedStudentsPanel.jsx:10]: i think preference is to use the style ben's using here. there's a couple of reasons i personally prefer this style:

1. this is how functional components are presented in the [react docs](https://reactjs.org/docs/components-and-props.html#function-and-class-components)
2. AFAIK it's more complicated to `export default` a named constant, and we want our components to always be named. example of what i mean:

```javascript
// exporting unnamed constant as default
export default (props) => {};

// exporting named constant as default
const MyComponent = props => {};
export default MyComponent;

// invalid js
export default const MyComponent = props => {};
```

File: apps/src/templates/codeReviewGroups/UnassignedStudentsPanel.jsx (hunk 3 above)
Comment (bencodeorg) [reply apps/src/templates/codeReviewGroups/UnassignedStudentsPanel.jsx:48]: Yeah good point, I'm exporting these constants and realizing we might just want to have a general `CodeReviewPanel` component that we can use for both sides of the UI (in which case, no exporting styles). I'm a little refactored out at the moment though :)

File: apps/src/templates/codeReviewGroups/Student.jsx (hunk 2 above)
Comment (bencodeorg) [reply apps/src/templates/codeReviewGroups/Student.jsx:61]: I think we'll probably update the navy to drag color that is a Code.org color at some point, going to punt this until then.

**Pull request merged**